lib/app.tsx

```diff
@@ -198,6 +199,7 @@ class AppComponent extends Component<Props> {
     return (
       <div className={appClasses}>
         {showEmailVerification && <EmailVerification />}
+        {showTokenLoginLogout && <TokenLoginLogout />}
```

Comment:
is it viable to leave this all in `boot` so that the magic link doesn't need to handled from "inside" the authenticated part of the app? or were you intentionally wanting to be able to show all the notes in the background with this modal? I was thinking that if we handle this in `boot` then we don't need a feedback loop for the token from inside the app and we can handle it all from a prompt, "Stay logged in as 'Dale'?   |   Login as 'Chip' instead"

if in-app, we're dropping the token altogether, right? it's just "Stay logged in as 'Dale'?   |   Logout"?

Reply:
I experimented a bit with this and I think I like it where it is. You're already logged in to this account, so it makes sense that it is a modal alert similar to the email verification one.

Correct that we're dropping the token in this case and just prompting the user to log out if desired. They would then have to manually enter the new username and password.